azuredevops/internal/service/wiki/resource_wiki_page.go

```diff
@@ -0,0 +1,169 @@
+package wiki
+
+import (
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/wiki"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+)
+
+/*
+To improve concurrent page api response:
+"The wiki page has already been updated by another client, so you cannot update it. Please try again."
+Add mutex lock to limit terraform provider concurrent create / update / delete request.
+*/
+var pageLock = sync.Mutex{}
```

Comment:
The lock can be replaced with `stateRefresh`. Ensure the resource is ready then we can continue on.
Ref: https://github.com/microsoft/terraform-provider-azuredevops/blob/main/azuredevops/internal/service/core/resource_project.go#L112

Reply:
A `StateChangeConf` instead is because API cannot ensure if the resource is created or sill in creating/updating status. `StateChangeConf` can be used to check the resource status in service side and ensure that the resource is create/update as our expectation then we can step forward.  Since the `mutex` can reach to the same target, it's OK to keep this implementation.